### DISCUSSION TITLE
Trigger a download process

### DISCUSSION THREAD

**Discussion by ramsy0dev** [reflex-dev/reflex#795] [Q&A] [answered]
I want to start a download process when the user clicks on the download button, but i don't know if using ```pc.link()``` and providing the direct download url into it as ```href``` will be a good idea, is there anything else i can do to accomplish this

**Comment by jq6l43d1** [comment, marked as answer]
I submitted an issue for this #909 The only ways I've seen to create a download link in pynecone are not very clean. Hopefully we'll have something like pc.download_files() or something in the future. I found a post on the pynecone discord server where someone was able to create download links like this:
```
@app.api.get("/download")
async def download_unmodified_video():
    # Find latest downloaded file
    file_path = max(glob.glob(f"{os.getcwd()}/downloads/**/*"))
    file_name = "unmodified_video.mp4"
    return FileResponse(path=file_path, media_type='video/mp4', filename=file_name)
.
.
.

pc.link("Download Me", href="http://localhost:8000/downloads")
```
I think there should be a better way.
